src/cli/gentest/request_manager.py

```diff
@@ -55,11 +55,11 @@ def eth_get_transaction_by_hash(self, transaction_hash: Hash) -> RemoteTransacti
         block_number = res.block_number
         assert block_number is not None, "Transaction does not seem to be included in any block"
 
-        assert (
-            res.ty == 0
-        ), f"Transaction has type {res.ty}: Currently only type 0 transactions are supported."
+        assert res.ty < 4, f"Unsupported transaction type :{res.ty}"
```

Comment:
I think this implies we support type 3 transactions too, and the only missing thing would be the versioned hashes.